troposphere/awslambda.py

```diff
@@ -4,11 +4,35 @@
 
 class Code(AWSProperty):
     props = {
-        'S3Bucket': (basestring, True),
-        'S3Key': (basestring, True),
+        'S3Bucket': (basestring, False),
+        'S3Key': (basestring, False),
         'S3ObjectVersion': (basestring, False),
+        'ZipFile': (basestring, False)
     }
 
+    def __init__(self, title=None, **kwargs):
+        super(Code, self).__init__(title, **kwargs)
+
+        if 'ZipFile' in kwargs:
```

Comment:
I would move this logic into validate() - the reason being that many times I've seen people create a troposphere resource/property, then modify it's attributes after the fact.  Validate() lets them do that still, without running into this.